core/src/core-api/auth.js

```diff
@@ -0,0 +1,31 @@
+// import * as AsyncHelpers from '../utilities/helpers/async-helpers';
+// import * as GenericHelpers from '../utilities/helpers/generic-helpers';
+import { config } from './config';
+
+class LuigiAuthManager {
+  constructor() { }
+
+  /*
+   * Detects if authorization is enabled via configuration.
+   * @returns {boolean} returns true if authorization is enabled. Otherwise returns false.
+   */
+  isAuthorizationEnabled() {
+    const idpProviderName = config.getConfigValue('auth.use');
+    const idpProviderSettings = config.getConfigValue(
+      `auth.${idpProviderName}`
+    );
+    return !!idpProviderSettings;
+  }
+
+  async handleAuthEvent(eventName, providerInstanceSettings, data, redirectUrl) {
+    const result = await config.executeConfigFnAsync('auth.events.' + eventName, providerInstanceSettings, data);
+    let redirect = result === undefined || !!result;
```

Comment:
Also, it will throw an exception in case of a rejected promise and the code after 'await' won't be executed.